osquery/tables/system/darwin/cpu_time.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+/*
+ *  Inspired from the psutil per_cpu_times function -
+ *  https://github.com/giampaolo/psutil/blob/ec1d35e41c288248818388830f0e4f98536b93e4/psutil/_psutil_osx.c#L739
+ */
+
+#include <mach/mach.h>
+#include <time.h>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+long int ticks_to_usecs(int ticks) {
+  return static_cast<long int>(
+      (static_cast<double>(ticks) / CLOCKS_PER_SEC * 1000000));
+}
+
+QueryData genCpuTime(QueryContext& context) {
+  QueryData results;
+
+  natural_t processor_count;
+  processor_cpu_load_info_data_t* processor_times;
+  mach_port_t host = mach_host_self();
+  mach_msg_type_number_t processor_msg_count;
+
+  kern_return_t ret =
+      host_processor_info(host,
```

Comment:
I'm worried in a few other locations there's a `vm_deallocate`. In the current form, this is not leaking memory, but to be safe let's deallocate the `processor_times` returned address:
- https://gist.github.com/mmisono/351441
- https://chromium.googlesource.com/external/webrtc/stable/src/+/cd52bf81d3586f5c9d576d2ddb52a488d5e485be/system_wrappers/source/cpu_mac.cc
- https://github.com/jlc/tmux-sys-stat/blob/master/cpureader_osx.cpp

If the response is a non-error.